lib/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -34,32 +34,46 @@
 	 ************************************/
 
 	/**
-	 * Get the presence state for this Channel.
+	 * Get the presence state for this Channel. Implicitly attaches the
+	 * Channel. However, if the channel is in or moves to the FAILED
+	 * state before the operation succeeds, it will result in an error
 	 * @return: the current present members.
 	 * @throws AblyException
 	 */
-	public synchronized PresenceMessage[] get()  {
+	public synchronized PresenceMessage[] get() throws AblyException {
+		if (channel.state == ChannelState.failed) {
```

Comment:
Currently, `channel.attach()` does not throw an exception for it. Would you like me to move the state check to `channel.attach()`, instead of `presence.get()`?